search-commons/src/test/java/no/unit/nva/search/ResourceClientTest.java

```diff
@@ -442,6 +448,168 @@ void searchWithUriReturnsOpenSearchAwsResponse(URI uri, int expectedCount) throw
         assertThat(pagedSearchResourceDto.totalHits(), is(equalTo(expectedCount)));
     }
 
+    @Test
+    void shouldFindAnthologyWithChapters() throws ApiGatewayException {
+        // Given that a parent document exists of type BookAnthology,
+        // and the parent document has two children of type AcademicChapter,
+        // when a query filters by hasParts of type AcademicChapter,
+        // then one document should be returned,
+        // and the returned document should be of type BookAnthology,
+        // and the returned document should have the ID of the parent document.
+
+        var uri =
+            UriWrapper.fromUri("https://x.org/?instanceTypeHasParts=AcademicChapter&from=0&size=20")
```

Comment:
so how does this test work with testdata? does it mock the response from the uri?

Reply:
All tests in ResourceClientTest uses the same opensearch instance, with the same content, no need for mocking nor populating between tests.